### PR TITLE
fabtests/pytest/conftest.py: changes to running remote commands

### PR DESCRIPTION
**Pull request by wenduwan**
Neuron runtime runs `lspci` in shell. This is a problem in the current implementation, which runs the remote command in non-interactive mode, e.g. `ssh 1.2.3.4 lspci` produces `sh: lspci: command not found`, because the non-interactive shell has a different environment. The solution is to run the command in interactive mode, i.e. `ssh 1.2.3.4 bash --login -c lspci`.
  However, please note the side effect that we are now locked into `bash`.

### CODE REVIEW

**Review by wzamazon — COMMENTED** [review]
Why not just pass LD_LIBRARY_PATH through `-E` option?

**Comment by wenduwan** [comment]
> Why not just pass LD_LIBRARY_PATH through `-E` option?

It doesn't actually work - dlopen happens before OOB exchange.
```
ssh -n -o StrictHostKeyChecking=no -o ConnectTimeout=30 -o BatchMode=yes 172.31.53.150 FI_EFA_USE_DEVICE_RDMA=1 timeout 360 /home/ec2-user/libfabric/fabtests/install/bin/fi_rdm_pingpong -I 5 -v -S r:0,4,64 -D neuron -p efa -E LD_LIBRARY_PATH=/opt/aws/neuron/lib:$LD_LIBRARY_PATH
[error] fabtests:common/hmem_neuron.c:85: Failed to dlopen libnrt.so.1
```

**Comment by wzamazon** [comment]
I meant to use the `-E` option of `runfabtests.py`. e.g.

`runfabtests.py -E "LD_LIBRARY_PATH=/opt/aws/neuron/lib FI_EFA_USE_DEVICE_RDMA=1"`

**Comment by wenduwan** [comment]
> I meant to use the `-E` option of `runfabtests.py`. e.g.
> 
> `runfabtests.py -E "LD_LIBRARY_PATH=/opt/aws/neuron/lib FI_EFA_USE_DEVICE_RDMA=1"`

I see. That's better. Thanks!

File: fabtests/pytest/conftest.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def populate_command(self, base_command, host_type, timeout=None):
             command = self._populate_ubertest_command(command, host_type)
         elif command.find("fi_multinode") != -1:
             command = self._populate_multinode_command(command, host_type)
-        else:
+        elif command.find("fi_") != -1:
```
Comment (wzamazon) [inline fabtests/pytest/conftest.py:105]: Not sure this is necessary. Does it cause any harm?

Comment (wzamazon) [reply fabtests/pytest/conftest.py:105]: There are some tests that does not start with `fi_`

Comment (wenduwan) [reply fabtests/pytest/conftest.py:105]: I see. Thanks for pointing that out.

It's not a bug. I wanted to reuse `populate_command` to execute `check_hmem` which does not support the same fabtest parameters.

I guess that's fine. I don't have to use this. Will update the change.

**Review by wzamazon — APPROVED** [review]

**Pull request merged**